crates/ruff_python_formatter/src/expression/string.rs

```diff
@@ -774,7 +781,22 @@ fn normalize_string(input: &str, quotes: StringQuotes, is_raw: bool) -> Cow<str>
 
     let mut chars = input.char_indices();
 
+    let is_raw = prefix.is_raw_string();
+    let is_fstring = prefix.is_fstring();
+    let mut in_formatted_value = false;
+
     while let Some((index, c)) = chars.next() {
+        if is_fstring && matches!(c, '{' | '}') {
+            if let Some(next) = input.as_bytes().get(index + 1).copied().map(char::from) {
```

Comment:
Should we make `chars` peekable?

Reply:
I don't have any evidence for this but I tried to avoid `peekable` because it leads to an additional write (to the peelable internal memory) and a conditional lookup, which can be expensive in a hot loop... But the compiler might be able to optimize it away. 

We could also use `Cursor` here which avoids this problem too